File: Programmers/Lv.1/실패율/42889.java
class Solution {
    public int[] solution(int N, int[] stages) {
        int[] answer = new int[N];
        
        double[] rates = new double[N];
        for (int i = 1; i <= N; i++) {
            int tryCount = 0;
            int failCount = 0;
            for (int stage : stages) {
                if (stage >= i) tryCount++;
                if (stage == i) failCount++;
            }
            
            answer[i - 1] = i;
            rates[i - 1] = (double) failCount / tryCount;
        }
        
        for (int a = 0; a < rates.length - 1; a++) {
            for (int b = a + 1; b < rates.length; b++) {
                if ((rates[a] < rates[b]) || (rates[a] == rates[b] && answer[a] > answer[b])) {
                    double rateTemp = rates[a];
                    rates[a] = rates[b];
                    rates[b] = rateTemp;
                    
                    int stageTemp = answer[a];
                    answer[a] = answer[b];
                    answer[b] = stageTemp;
                }
            }
        }
        
        return answer;
    }
}

File: Programmers/Lv.1/실패율/42889.js
function solution(N, stages) {
    const answer = [];
    
    const arr = [];
    for (let i = 1; i <= N; i++) {
        let tryCount = 0;
        let failCount = 0;
        stages.forEach(v => {
            if (v >= i) tryCount++;
            if (v == i) failCount++;
        });
        
        arr.push([i, failCount / tryCount]);
    }
    
    arr.sort((a, b) => b[1] - a[1]).forEach(v => answer.push(v[0]));
    
    return answer;
}
